File: testers/avro/java-vanilla/generated-and-checked-in/com/example/events/serde/OrderUpdatedSerde.java
package com.example.events.serde;

import com.example.events.OrderUpdated;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.util.Map;
import org.apache.avro.generic.GenericDatumReader;
import org.apache.avro.generic.GenericDatumWriter;
import org.apache.avro.generic.GenericRecord;
import org.apache.avro.io.BinaryDecoder;
import org.apache.avro.io.BinaryEncoder;
import org.apache.avro.io.DatumReader;
import org.apache.avro.io.DatumWriter;
import org.apache.avro.io.DecoderFactory;
import org.apache.avro.io.EncoderFactory;
import org.apache.kafka.common.errors.SerializationException;
import org.apache.kafka.common.serialization.Deserializer;
import org.apache.kafka.common.serialization.Serde;
import org.apache.kafka.common.serialization.Serializer;

/** Serde for OrderUpdated */
public class OrderUpdatedSerde
    implements Serde<OrderUpdated>, Serializer<OrderUpdated>, Deserializer<OrderUpdated> {
  DatumWriter<GenericRecord> writer = new GenericDatumWriter(OrderUpdated.SCHEMA);

  DatumReader<GenericRecord> reader = new GenericDatumReader(OrderUpdated.SCHEMA);

  @Override
  public void configure(Map<String, ?> configs, boolean isKey) {}

  @Override
  public byte[] serialize(String topic, OrderUpdated data) {
    if (data == null) {
      return null;
    }
    try {
      ByteArrayOutputStream out = new ByteArrayOutputStream();
      BinaryEncoder encoder = EncoderFactory.get().binaryEncoder(out, null);
      writer.write(data.toGenericRecord(), encoder);
      encoder.flush();
      return out.toByteArray();
    } catch (IOException e) {
      throw new SerializationException("Error serializing Avro message", e);
    }
  }

  @Override
  public OrderUpdated deserialize(String topic, byte[] data) {
    if (data == null) {
      return null;
    }
    try {
      BinaryDecoder decoder = DecoderFactory.get().binaryDecoder(data, null);
      GenericRecord record = reader.read(null, decoder);
      return OrderUpdated.fromGenericRecord(record);
    } catch (IOException e) {
      throw new SerializationException("Error deserializing Avro message", e);
    }
  }

  @Override
  public void close() {}

  @Override
  public Serializer<OrderUpdated> serializer() {
    return this;
  }

  @Override
  public Deserializer<OrderUpdated> deserializer() {
    return this;
  }
}

File: testers/avro/kotlin/generated-and-checked-in/com/example/events/serde/OrderUpdatedSerde.kt
package com.example.events.serde

import com.example.events.OrderUpdated
import io.confluent.kafka.serializers.KafkaAvroDeserializer
import io.confluent.kafka.serializers.KafkaAvroSerializer
import kotlin.collections.MutableMap
import org.apache.avro.generic.GenericRecord
import org.apache.kafka.common.serialization.Deserializer
import org.apache.kafka.common.serialization.Serde
import org.apache.kafka.common.serialization.Serializer

/** Serde for OrderUpdated */
class OrderUpdatedSerde() : Serde<OrderUpdated>, Serializer<OrderUpdated>, Deserializer<OrderUpdated> {
  override fun close() {
    innerSerializer.close()
    innerDeserializer.close()
  }

  override fun configure(
    configs: MutableMap<kotlin.String, *>,
    isKey: kotlin.Boolean
  ) {
    innerSerializer.configure(configs, isKey)
    innerDeserializer.configure(configs, isKey)
  }

  override fun deserialize(
    topic: kotlin.String,
    data: ByteArray?
  ): OrderUpdated? {
    if (data == null) {
      return null
    }
    val record: GenericRecord = (innerDeserializer.deserialize(topic, data) as GenericRecord)
    return OrderUpdated.fromGenericRecord(record)
  }

  override fun deserializer(): Deserializer<OrderUpdated> = this

  val innerDeserializer: KafkaAvroDeserializer = KafkaAvroDeserializer()

  val innerSerializer: KafkaAvroSerializer = KafkaAvroSerializer()

  override fun serialize(
    topic: kotlin.String,
    data: OrderUpdated?
  ): ByteArray? {
    if (data == null) {
      return null
    }
    return innerSerializer.serialize(topic, data.toGenericRecord())
  }

  override fun serializer(): Serializer<OrderUpdated> = this
}